superset/views/geocoding.py

```diff
@@ -134,76 +137,66 @@ def geocode(self) -> Response:
                  or an error message if somethings went wrong
         """
         request_data = request.json
-        columns = []
-
         self._set_geocoder(request_data.get("geocoder", ""))
-        if request_data.get("streetColumn"):
-            columns.append(request.json.get("streetColumn"))
-        if request_data.get("cityColumn"):
-            columns.append(request.json.get("cityColumn"))
-        if request_data.get("countryColumn"):
-            columns.append(request.json.get("countryColumn"))
-        if request_data.get("buildingNumberColumn"):
-            columns.append(request_data.get("buildingNumberColumn"))
-        if_exists = request_data.get("overwriteIfExists")
         lat_column = request_data.get("latitudeColumnName", "lat")
         lon_column = request_data.get("longitudeColumnName", "lon")
-        save_on_stop_geocoding = request.json.get("saveOnErrorOrInterrupt", True)
+        columns = self._create_column_list(request_data)
+        if_exists = request_data.get("ifExists")
+        save_on_stop_geocoding = request_data.get("saveOnErrorOrInterrupt", True)
         table_dto = request_data.get("datasource", models.TableDto())
         table_id = table_dto.get("id", "")
         message_suffix = ""
         try:
-            column_message = self._check_and_create_columns(request_data)
-            message_suffix = f" but {column_message} please choose the overwrite option when trying again"
-
-            if "append" in if_exists:
-
-                table_data = self._load_data_from_columns(
-                    table_id, columns, lat_column, lon_column
-                )
-            else:
-                table_data = self._load_data_from_columns(table_id, columns)
-
-        except ValueError as e:
-            self.logger.exception(f"ValueError when querying for lat/lon columns {e}")
-            self.stats_logger.incr("geocoding_failed")
-            return json_error_response(f"{e.args[0]} {message_suffix}", status=400)
-        except SqlSelectException as e:
-            self.logger.exception(
-                f"SqlSelectException when preparing for geocoding {e.orig}"
+            table = self._get_table_with_columns(table_id)
+            lat_exists = self._does_column_name_exist(table, lat_column)
+            lon_exists = self._does_column_name_exist(table, lon_column)
+            if "fail" in if_exists and (lat_exists or lon_exists):
+                self.stats_logger.incr("geocoding_failed")
+                return json_error_response("At least one of the columns already exists!", status=400)
+            self._create_columns(
+                lat_column, lat_exists, lon_column, lon_exists, table
             )
+            message_suffix = f" but columns have been created please choose the 'Replace' or 'Append' option when trying again."
```

Comment:
what if both columns already exist? And also if replace/append was chosen we still get this message?

Reply:
A lot of code was written only for this message, all the user needs to know is that the "Fail" option is no good anymore